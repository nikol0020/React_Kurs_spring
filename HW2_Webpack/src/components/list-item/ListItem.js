import React, {Component} from 'react';
import './ListItem.scss';

import PropTypes from 'prop-types';

const ListItem = (props) => {

    const {id, title, genres, release_date, poster_path} = props;

    return (

        <div className="list-item" key={ id }>
            <img src={poster_path} alt="picture" className="list-item__image"/>
            <div className="list-item__short-description">
                <h4 className="list-item__title">{ title }</h4>
                <span className="list-item__genre">{ genres }</span>
                <span className="list-item__year">{ release_date }</span>
            </div>
        </div>
    )
};

ListItem.propTypes = {
    id: PropTypes.number,
    title: PropTypes.string,
    genres: PropTypes.string,
    release_date: PropTypes.string,
    poster_path: PropTypes.string
};

export default ListItem;
