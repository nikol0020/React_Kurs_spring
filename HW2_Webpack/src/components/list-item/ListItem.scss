.list-item {
  width: 200px;
  height: 430px;
  background-color: #fff;
  margin: 20px;
  color: #333;
  border: 2px solid #a3a;

  &__title {
    margin: 10px 0;
    text-align: center;
  }

  &__image {
    width: 100%;
    height: 315px;
    min-width: 200px;
    min-height: 300px;
    background-color: #bbf;
  }

  &__short-description {
    margin: 10px;
  }

  &__genre {
    float: left;
    overflow: hidden;
    text-overflow: ellipsis;
    height: 4.5vh;
  }

  &__year {
    float: right;
    color: #a3a;
  }

}
