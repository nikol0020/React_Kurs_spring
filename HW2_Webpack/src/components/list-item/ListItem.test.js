import React from 'react';
import ListItem from './ListItem';
import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';

describe('<ListItem /> shallow rendering', () => {

    it('should render div', () => {
        const wrapper = shallow(
            <div className="list-item" ></div>
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('should render a list-item__image', () => {
        const wrapper = shallow(
            <img poster_path="poster_path" />
        );
        expect(wrapper.prop('poster_path')).toEqual('poster_path');
    });

    it('should render a list-item__title', () => {
        const wrapper = shallow(
            <h4 title="title" />
        );
        expect(wrapper.prop('title')).toEqual('title');
    });

    it('should render a list-item__year', () => {
        const wrapper = shallow(
            <span release_date="release_date"></span>
        );
        expect(wrapper.prop('release_date')).toEqual('release_date');
    });

});