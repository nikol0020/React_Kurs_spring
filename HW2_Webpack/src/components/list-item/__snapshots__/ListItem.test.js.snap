// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListItem /> shallow rendering should render div 1`] = `
<div
  className="list-item"
/>
`;
